compile.py

```diff
@@ -4,13 +4,16 @@
 """Creates a long file of yearly Journal Impact Factors."""
 
 from pathlib import Path
-
 import pandas as pd
-from tqdm import tqdm
+import requests
+from io import StringIO
+import time
 
```

Comment:
Please group and order imports according to PEP8 (https://peps.python.org/pep-0008/#imports): 
> Imports should be grouped in the following order:
>
>Standard library imports.
>Related third party imports.
>Local application/library specific imports.

Within each group, first the full imports, then the partial imports. Each sorted alphabetically.